src/components/Items/ItemViewer.tsx

```diff
@@ -0,0 +1,298 @@
+import React, {useState, useEffect} from 'react';
+import {orderBy, some, map, filter} from 'lodash'
+import './ItemViewer.css';
+import {
+    MDBDataTable,
+    MDBContainer,
+    MDBView,
+    MDBMask,
+    MDBRow,
+    MDBCol
+} from 'mdbreact';
+import maths from '../../shared/utilities/maths'
+import Search from '../../shared/components/Search'
+import Highlight from "../../shared/components/Highlight";
+import {useSelector} from "react-redux";
+import {GlobalState} from "../../store/store";
+import Item, {ItemProperty} from "./Item";
+import CSS from 'csstype';
+
+// Ce compossant fait tellement trop de choses, je gagnerais à le découper.
+const ItemViewer = () =>
```

Comment:
This class is doing too much things, I should try to split it :
- display item, item attributes, item name
- calcul damage, defence, attributes, item image style. I think most of this part should be handled by backend.
- Generate columns, generate rows.
- Display dataTable, search, layout, too much things.